others/serialization.cpp

```diff
@@ -0,0 +1,163 @@
+/**
+ * @file
+ * @brief A simple Serializer and Deserializer utility for fundamental data
+ * types and strings.
+ */
+#include <cassert>      // for assert
+#include <cstdint>      // for std::uint32_t
+#include <fstream>      // for std::ifstream std::ofstream
+#include <iostream>     // for std::ios std::cout std::cerr
+#include <string>       // for std::string
+#include <type_traits>  // for std::is_fundamental
+
+/**
+ * @brief A utility class for serializing fundamental data types and strings to
+ * a binary file.
```

Comment:
```suggestion
 * a binary file.
 * @class Serializer
```

Reply:
I dont understand this one, what exactly i have to do here?